### ISSUE TITLE
use DSP's conv for 0.6

### ISSUE THREAD

**Issue by HDictus** [JuliaDSP/DSP.jl#271]
Unless I've misunderstood, for version  > "0.7.0-DEV.986" we don't actually use the conv deconv, filt, filt! and xcorr defined in DSP.jl, but load them from julia and ignore our local version
Is there a  reason for this restriction? 
Presently, implementing new behavior for conv and testing it results in failed tests in CI, one could just put an if-end around the tests in question, but it'd be preferable to keep everything the same for different versions (say, for example some function that isn't imported from base in 0.6 develops a dependency on a new behavior of conv/xcorr/deconv/filt).

On a related note, is it really a good idea to test for compatibility with julia nightly, considering it is unstable, and may fail for reasons unrelated to the PR? Perhaps we should just test with full releases?

**Comment by martinholters** [comment]
I don't think we should overwrite any `conv` etc. methods from `Base` (on 0.6), so that would leave the option of having the `Base` functions and the `DSP` ones be separate on 0.6 (and don't have DSP `export` them). Then one has to decide which one to bind to with `using` (or use explicit qualification). My preferred solution here would be to drop support for 0.6 altogether.

But I do find testing on Julia nightly useful. True, it may sometimes cause unrelated failures. But it may also catch issues early in case a breaking change slips through unnoticed in Julia.

**Comment by HDictus** [comment]
I'd be cool with dropping 0.6, didn't realize that was an option.

**Comment by ssfrr** [comment]
Yeah, I think dropping 0.6 for future releases of DSP is reasonable. There are a number of breaking changes in the wings, so I think the set of folks willing to put up with breaking DSP changes but stuck on Julia 0.6 is probably pretty small.

**Comment by HDictus** [comment]
Where can we specify the commit/version/branch for julia 0.6 to use and install? Does the package listing handle it for us based on REQUIRE?

**Comment by ssfrr** [comment]
I might be misunderstanding your question, but `REQUIRE` lets you specify bounds on the julia version that a package supports. It won't install a different version of Julia when you try to install the package, but it will only let you install packages that are compatible with your current Julia version. If someone installs DSP on Julia 0.6 they will get the most recent release that is compatible with their Julia version.

**Comment by HDictus** [comment]
I think you did misunderstand my question, but your clarification answered it anywa, thanks!

**Comment by galenlynch** [comment]
Is this resolved?

**Comment by HDictus** [comment]
If you're all happy with #272, then yes
